lib/services/local/presence.go

```diff
@@ -714,7 +714,7 @@ Acquire:
 			// however, to assume that we are under high contention and attempt to
 			// spread out retries via random backoff.
 			select {
-			case <-time.After(s.jitter(baseBackoff * time.Duration(i))):
+			case <-time.After(s.jitter(baseBackoff * time.Duration(acquireAttempts))):
```

Comment:
This seems to randomize the jitter in a more favorable distribution - otherwise the concurrent process jitter in lock-step with values close to each other. On my machine this bumped the max number of concurrent requests to over 70 (although the docs refer to 100 which I was not able to confirm).
I'm sure there are additional tweaks that can help.